lib/spack/docs/tutorial_modules.rst

```diff
@@ -1,52 +1,278 @@
-something like this:
+The fastest way to set-up your environment is to :ref:`use a Docker image <workflow_create_docker_image>`:
+
+.. code-block:: console
+
+  $ docker pull alalazo/spack:module_tutorial
+  $ docker run --rm -h module-file-tutorial -it alalazo/spack:module_tutorial
+  root@module-file-tutorial:/#
+
+If you arrived at this point you should be ready to start, as all the software needed is
+pre-installed in the image:
+
+.. code-block:: console
+
+  root@module-file-tutorial:/# which spack
+  /usr/local/bin/spack
+  root@module-file-tutorial:/# spack find
+  ==> 46 installed packages.
+  -- linux-ubuntu16.04-x86_64 / gcc@5.4.0 -------------------------
+  autoconf@2.69    gcc@7.2.0  git@2.9.4  isl@0.18         libtool@2.4.6  lua@5.3.4                lua-luaposix@33.4.0  mpc@1.0.3   ncurses@6.0  pkg-config@0.29.2  tcl@8.6.6
+  automake@1.15.1  gdbm@1.13  gmp@6.1.2  libsigsegv@2.11  lmod@7.7       lua-luafilesystem@1_6_3  m4@1.4.18            mpfr@3.1.5  perl@5.24.1  readline@7.0       zlib@1.2.11
+
+  -- linux-ubuntu16.04-x86_64 / gcc@7.2.0 -------------------------
+  bzip2@1.0.6  ncurses@6.0             netlib-scalapack@2.0.2  openblas@0.2.20  pkg-config@0.29.2  py-packaging@16.8   py-setuptools@35.0.2  readline@7.0
+  cmake@3.9.4  netlib-lapack@3.6.1     netlib-scalapack@2.0.2  openmpi@1.10.2   py-appdirs@1.4.3   py-pyparsing@2.2.0  py-six@1.10.0         sqlite@3.20.0
+  mpich@3.2    netlib-scalapack@2.0.2  netlib-scalapack@2.0.2  openssl@1.0.2k   py-numpy@1.13.1    py-scipy@0.19.1     python@2.7.14         zlib@1.2.11
+
+Go to :ref:`module_file_tutorial_non_hierarchical` to proceed with the tutorial.
+
+.. note::
+  Dockerfile for this image
+    Those of you that want to build a similar container themselves can find the
+    ``Dockerfile`` and the other resources in Spack's ``share/spack/docs/docker``
+    folder.
+
+.. _module_file_tutorial_work_locally:
+
+^^^^^^^^^^^^^^^^^^^^^^
+Work in a local folder
+^^^^^^^^^^^^^^^^^^^^^^
+
+If you don't feel like using a container, you can set-up your environment
+locally. Let's start by cloning Spack repository and moving to the directory
+where it was checked out:
+
+.. code-block:: console
+
+  $ git clone https://github.com/LLNL/spack.git
+  $ cd spack
+
+From here we'll be building up the required stack of software.
+
+"""""""""""""""""""
+Build a module tool
+"""""""""""""""""""
+
+The first thing that we need is the module tool. In this case we
+choose ``lmod`` as it can work with both hierarchical and non-hierarchical
+module file layouts.
+
+.. code-block:: console
+
+  $ bin/spack install lmod
+  ...
+  output omitted
+  ...
+
+Once the module tool is installed we need to have it available in the
+current shell. As the installation directories are definitely not easy
+to remember, we'll employ the command ``spack location`` to retrieve the
+``lmod`` prefix directly from Spack:
+
+.. code-block:: console
+
+  $ . $(spack location -i lmod)/lmod/lmod/init/bash
```

Comment:
I think there is a `profile` script in the same directory that should be more generic. Just in case we have any zsh users 😉

Reply:
I hope at SC17 we'll be using the docker image :smile: